adaptor-rm-webapp/src/main/java/com/sample/rm/RMToolManager.java

```diff
@@ -58,34 +63,56 @@ private static int randomNumber(int origin, int bound) {
 		return ThreadLocalRandom.current().nextInt(origin, bound);
 	}
 
-	private static Requirement createRandomRequirement(String id) {
-		Requirement r = null;
-		try {
-			r = RMToolResourcesFactory.createRequirement(id);
-			r.setTitle("A sample Requirement with id:" + id);
-			r.setDescription("A sample Requirement with id:" + id);
-		} catch (URISyntaxException e) {
+    public static ArrayList<Requirement> getRequirements() {
+        return new ArrayList<Requirement>(requirements.values());
+    }
+
+    private static Requirement produceRandomRequirement(String id) {
+        Requirement r = null;
+        try {
+            r = RMToolResourcesFactory.createRequirement(id);
+            r.setIdentifier(id);
+            r.setTitle("aTitle with id:" + id);
+            r.setDescription("A sample Requirement with id:" + id);
+        } catch (URISyntaxException e) {
             log.error("Failed to create resource", e);
-		}
-		return r;
-	}
+        }
+        return r;
+    }
 
-	private static List<Requirement> createRandomRequirements(int min, int max, int minId, int maxId)
-    {
-		int size = randomNumber(min, max);
-		List<Requirement> resources = new ArrayList<Requirement>(size);
-		for (int i = 0; i < size; i++) {
-	    	resources.add(createRandomRequirement(Integer.toString(randomNumber(minId, maxId))));
-		}
-        return resources;
+    private static void initializeRequirements(int size) {
+        for (int i = 0; i < size; i++) {
+            nextRequirementId++;
+            String id = Integer.toString(nextRequirementId);
+            Requirement r = produceRandomRequirement(id);
+            requirements.put(id, r);
+        }
+    }
+
+    public static Requirement createOrUpdateRequirement(HttpServletRequest httpServletRequest, final Requirement aResource) {
+        if (!requirements.containsKey(aResource.getIdentifier())) {
+            nextRequirementId++;
+            String id = Integer.toString(nextRequirementId);
+            URI uri = RMToolResourcesFactory.constructURIForRequirement(id);
+            aResource.setAbout(uri);
+            requirements.put(id, aResource);
+			InmemPagedTrsSingleton.getInmemPagedTrs().onCreated(aResource);
+        }
+        else {
+            URI uri = RMToolResourcesFactory.constructURIForRequirement(aResource.getIdentifier());
+            aResource.setAbout(uri);
+            requirements.put(aResource.getIdentifier(), aResource);
+			InmemPagedTrsSingleton.getInmemPagedTrs().onModified(aResource);
```

Comment:
- users shall be able to replce the implementation
- in-mem implementation implements two interfaces in one class. our code needs to use two interfaces as if they were totally separate